src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -82,34 +93,38 @@ const CheckboxInput: FC<Props> = ({
     styles.checkbox,
     className,
     { [styles.disabled]: isDisabled },
+    { [styles.inline]: displayInline },
   );
 
+  const inputProps = {
+    'aria-invalid': !!error,
+    'aria-label': label,
+    'aria-labelledby': label ? `${id}Label` : undefined,
+    id,
+    checked: !!isChecked,
+    disabled: isDisabled,
+    onBlur: handleBlur,
+    onChange: handleChange,
+    onFocus: handleFocus,
+    type: 'checkbox',
+    className: styles.input,
+  };
+
+  const labelProps = {
+    isFieldRequired: isRequired,
+    inputId: id,
+    hasError: !!error,
+    isDisabled,
+  };
+
   return (
     <>
       <div className={wrapperClasses}>
-        <input
-          aria-invalid={!!error}
-          aria-label={label}
-          aria-labelledby={label ? `${id}Label` : undefined}
-          id={id}
-          checked={!!isChecked}
-          disabled={isDisabled}
-          onBlur={handleBlur}
-          onChange={handleChange}
-          onFocus={handleFocus}
-          type="checkbox"
-          className={styles.input}
-        />
-        {label && (
-          <FormLabel
-            {...{
-              isFieldRequired: isRequired,
-              inputId: id,
-              labelText: label,
-              hasError: !!error,
-              isDisabled,
-            }}
-          />
+        <input {...inputProps} />
+        {label && !hideLabel && (
+          <FormLabel {...labelProps}>
```

Comment:
props spreading seems unnecessary here and makes it harder to read in my opinion